featuretools/feature_base/feature_base.py

```diff
@@ -366,16 +371,45 @@ class DirectFeature(FeatureBase):
     input_types = [Variable]
     return_type = None
 
-    def __init__(self, base_feature, child_entity):
+    def __init__(self, base_feature, child_entity=None, relationship_path=None):
```

Comment:
for discussion: 

I think this extends the API for direct features. Previously the feature had to come from a parent, but with a relationship path it can no come from a grandparent, great grandparent, etc. 

I don't see anything wrong with this good change, but it does have implications on feature calculation in pandas backend

Reply:
Would it be better to keep the current behavior for now (only allow paths of length 1), and leave this enhancement for a future PR? Just to avoid the scope of the current project too much. On the other hand since we are refactoring so much it might be less overall work to build in support now.

Reply:
what if we leave support for creating such features now, but pandas backend need to get support for calculating them. that doesn't really change the status quo